### PR TITLE
cluster template improvements

### PR DESCRIPTION
**Pull request by tuxtof**
**What this PR does / why we need it**:

- upgrade kubevip to v0.5.5
- Allow to enable Load Balancing SVC Load Balancers
- Implement tls-cipher-suites enforcement
- fix eviction-hard for multi master

**How Has This Been Tested?**:

Manual test + E2E test
```
Ran 12 of 23 Specs in 1464.550 seconds
SUCCESS! -- 12 Passed | 0 Failed | 0 Pending | 11 Skipped
PASS
```

**Release note**:

```release-note
- upgrade kubevip to v0.5.5
- Allow to enable Load Balancing SVC Load Balancers
- Implement tls-cipher-suites enforcement
- fix eviction-hard for multi master
```

### CODE REVIEW

File: templates/base/cluster-template.yaml [hunk 1]
```diff
@@ -116,9 +116,15 @@ spec:
           - localhost
           - 127.0.0.1
           - 0.0.0.0
+        extraArgs:
```
Comment (deepakm-ntnx) [inline templates/base/cluster-template.yaml:119]: is this required for kube-vip 5.5 upgrade?

File: templates/base/cluster-template.yaml [hunk 2]
```diff
@@ -160,6 +166,8 @@ spec:
                     value: "${KUBEVIP_SVC_ENABLE=false}"
                   - name: lb_enable
                     value: "${KUBEVIP_LB_ENABLE=false}"
+                  - name: enableServicesElection
```
Comment (deepakm-ntnx) [inline templates/base/cluster-template.yaml:169]: why is this required?

File: templates/base/cluster-template.yaml (hunk 1 above)
Comment (tuxtof) [reply templates/base/cluster-template.yaml:119]: no this is for `Implement tls-cipher-suites enforcement`

File: templates/base/cluster-template.yaml (hunk 2 above)
Comment (tuxtof) [reply templates/base/cluster-template.yaml:169]: this is for `Allow to enable Load Balancing SVC Load Balancers`

**Review by deepakm-ntnx — DISMISSED** [review]

**Review by yannickstruyf3 — APPROVED** [review]

File: templates/base/cluster-with-kcp.yaml [hunk 3]
```diff
@@ -59,9 +59,15 @@ spec:
           - localhost
           - 127.0.0.1
           - 0.0.0.0
+        extraArgs:
+          tls-cipher-suites: "${TLS_CIPHER_SUITES=TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256,TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256,TLS_ECDHE_ECDSA_WITH_AES_256_GCM_SHA384,TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384,TLS_ECDHE_ECDSA_WITH_CHACHA20_POLY1305_SHA256,TLS_ECDHE_RSA_WITH_CHACHA20_POLY1305_SHA256}"
```
Comment (deepakm-ntnx) [inline templates/base/cluster-with-kcp.yaml:63]: where did we get this recommended list from? why so many? can we pick the most secure one for each k8s version?

**Review by deepakm-ntnx — APPROVED** [review]

File: templates/base/cluster-with-kcp.yaml (hunk 3 above)
Comment (tuxtof) [reply templates/base/cluster-with-kcp.yaml:63]: this is the commonly accepted as secure
same used by Openshift

Comment (deepakm-ntnx) [reply templates/base/cluster-with-kcp.yaml:63]: https://github.com/kubernetes/kubernetes/issues/83669
